lib/services/role.go

```diff
@@ -1979,6 +1995,36 @@ func (set RoleSet) EnhancedRecordingSet() map[string]bool {
 	return m
 }
 
+func (set RoleSet) CreateHostUser(s types.Server) bool {
+	for _, role := range set {
+		result, _, err := MatchLabels(role.GetNodeLabels(types.Allow), s.GetAllLabels())
+		if err != nil {
+			return false
+		}
+		// skip nodes that dont have matching labels
+		if !result {
+			continue
+		}
+
+		createHostUser := role.GetOptions().CreateHostUser
+
+		// if any of the matching roles do not enable create host
+		// user, the user should not be allowed on
+		if createHostUser != nil && createHostUser.Value {
+			return false
+		}
+	}
+	return true
+}
+
+func (set RoleSet) HostGroups() []string {
+	groups := []string{}
+	for _, role := range set {
+		groups = append(groups, role.GetHostGroups(types.Allow)...)
```

Comment:
This should only return host groups for the roles matching the server, not all roles in the set.